### PR TITLE
Wip/clippy fixes

### PR DESCRIPTION
**Pull request by hunger**
The daily dosage of clippy fixes.

### CODE REVIEW

File: sixtyfps_compiler/passes/z_order.rs [hunk 1]
```diff
@@ -41,9 +41,7 @@ fn reorder_children_by_zorder(
             .remove("z")
             .and_then(|e| eval_const_expr(&e.expression, "z", &e, diag));
         let z = z.or_else(|| {
-            if child_elm.borrow().repeated.is_none() {
-                return None;
-            };
+            child_elm.borrow().repeated.as_ref()?;
```
Comment (tronical) [inline sixtyfps_compiler/passes/z_order.rs:44]: I really like every part of this patch except this one :-). Is it just me?

Comment (hunger) [reply sixtyfps_compiler/passes/z_order.rs:44]: It definitely is canonical rust: Clippy explicitly says so:-)

The justification for this clippy lint is this: "Question mark usage is more idiomatic."

But I do not mind to just disable `clippy::question_mark`: I have never had this one before indicate some typo/copy and paste error.

Comment (tronical) [reply sixtyfps_compiler/passes/z_order.rs:44]: I’d favor disabling it in this case, but not too strong feeling about it. @ogoffart what’s your opinion?

Comment (ogoffart) [reply sixtyfps_compiler/passes/z_order.rs:44]: I like canonical rust ...  What's wrong with it?

**Review by tronical — APPROVED** [review]

**Pull request merged**